net/dns/resolver/forwarder_test.go

```diff
@@ -400,11 +398,24 @@ func runDNSServer(tb testing.TB, opts *testDNSServerOptions, response []byte, on
 	return
 }
 
+var (
+	enableDebugOnce sync.Once
+	forceDebug      atomic.Bool
+)
+
 func enableDebug(tb testing.TB) {
-	const debugKnob = "TS_DEBUG_DNS_FORWARD_SEND"
-	oldVal := os.Getenv(debugKnob)
-	envknob.Setenv(debugKnob, "true")
-	tb.Cleanup(func() { envknob.Setenv(debugKnob, oldVal) })
+	forceDebug.Store(true)
+	tb.Cleanup(func() { forceDebug.Store(false) })
+
+	enableDebugOnce.Do(func() {
+		oldFunc := verboseDNSForward
+		verboseDNSForward = func() bool {
```

Comment:
now we only overwrite this global once for the first test that wants it, but the tests actually just use the `forceDebug` atomic instead

Reply:
Yuck, but at least it's self-contained 😆